cmd/operator-sdk/bundle/create_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/bundle"
+)
+
+var _ = Describe("Running a bundle validate command", func() {
+	Describe("NewCreateCmd", func() {
+		It("builds and returns a cobra command", func() {
+			cmd := bundle.NewCreateCmd()
+			Expect(cmd).NotTo(BeNil())
+			Expect(cmd.Use).To(Equal("create"))
+			Expect(cmd.Short).To(Equal("Create an operator bundle image"))
+			Expect(cmd.Long).To(Equal(`The 'operator-sdk bundle create' command will build an operator
+bundle image containing operator metadata and manifests, tagged with the
```

Comment:
@estroz do you think that we should check if the cmd help string is ==? 
IMO I think we should test the command by verifying that it works as expected instead of it. 
Example:
operator-sdk bundle create
- Should successfully work when the flag --generated-only is informed and then should create in the disk a bundle with the files X. and then, we should ensure that the expected files were generated successfully. 
- Should fail when the crds are not found 
- Should fail when is not executed from the root path

Reply:
these are intended to just unit test the cmd layer, not the entire command. So, testing that args are parsed and help text is output correctly, not to test the entire e2e functionality of the command, such as actually generating the output and writing to disk. Once KB integration has happened, I'd like to refactor the interface between the cmd parts and the pkg libs that actually do the stuff, and then add unit tests that ensure the correct methods are called with the right info, which is what the pending Run tests up above will be.

Reply:
Personally I don't see any benefit from comparing actual vs. expected command help strings. With these comparisons we'd have to update help text in 2 places. Perhaps `strings.TrimSpace(cmd.Long) != ""` but not the entire text.

Reply:
Yeah, I was just trying to be complete. If I had TDD'd this package from the start, the tests would be the definitive source of the content, so I was just trying to replicate that. I can omit them if you think that's too cumbersome.

Reply:
IMO we do not need to these checks at all. See that the tests added here did not change the % of coverage and are not testing the commands. Just if they have usage, long desc and etc.

Reply:
These tests are covering the lines that set the long, usage, etc. for the command. I strongly think we should have a test that tests that the fields were populated correctly, but am open to suggestions for something more clever than Equal(). I'm not sure what estroz meant with the TrimSpace thing, doesn't that just trim leading and trailing whitespace?

Reply:
Really tks for all your help closing it assorted out since we spoke today and agreed in the format to move forward.